modules/base/rendering/renderabletrail.cpp

```diff
@@ -23,346 +23,355 @@
+    , _pointSize("pointSize", "Point Size", 1, 1, 64)
+    , _renderingModes(
+        "renderingMode",
+        "Rendering Mode",
+        properties::OptionProperty::DisplayType::Dropdown
+    )
 {
-    _successfullDictionaryFetch &= dictionary.getValue(keyBody, _target);
-    _successfullDictionaryFetch &= dictionary.getValue(keyObserver, _observer);
-    _successfullDictionaryFetch &= dictionary.getValue(keyFrame, _frame);
-    _successfullDictionaryFetch &= dictionary.getValue(keyTropicalOrbitPeriod, _tropic);
-    _successfullDictionaryFetch &= dictionary.getValue(keyEarthOrbitRatio, _ratio);
-    _successfullDictionaryFetch &= dictionary.getValue(keyDayLength, _day);
-    // values in modfiles set from here
-    // http://nssdc.gsfc.nasa.gov/planetary/factsheet/marsfact.html
-
-    glm::vec3 color(0.f);
-    if (dictionary.hasKeyAndValue<glm::vec3>(keyColor))
-        dictionary.getValue(keyColor, color);
-    _lineColor = color;
-
-    bool timeStamps = false;
-    if (dictionary.hasKeyAndValue<bool>(keyStamps))
-        dictionary.getValue(keyStamps, timeStamps);
-    _showTimestamps = timeStamps;
-    addProperty(_showTimestamps);
-
-    _lineColor.setViewOption(properties::Property::ViewOptions::Color);
+    _translation = std::unique_ptr<Translation>(Translation::createFromDictionary(
+        dictionary.value<ghoul::Dictionary>(KeyTranslation)
+    ));
+
+    _lineColor = dictionary.value<glm::vec3>(KeyColor);
     addProperty(_lineColor);
 
+    if (dictionary.hasKeyAndValue<bool>(KeyEnableFade)) {
+        _useLineFade = dictionary.value<bool>(KeyEnableFade);
+    }
+    addProperty(_useLineFade);
+
+    if (dictionary.hasKeyAndValue<double>(KeyFade)) {
+        _lineFade = dictionary.value<double>(KeyFade);
+    }
     addProperty(_lineFade);
 
+    if (dictionary.hasKeyAndValue<double>(KeyLineWidth)) {
+        _lineWidth = dictionary.value<double>(KeyLineWidth);
+    }
     addProperty(_lineWidth);
-    _distanceFade = 1.0;
-}
 
-bool RenderableTrail::initialize() {
-    if (!_successfullDictionaryFetch) {
-        LERROR("The following keys need to be set in the Dictionary. Cannot initialize!");
-        LERROR(keyBody << ": " << _target);
-        LERROR(keyObserver << ": " << _observer);
-        LERROR(keyFrame << ": " << _frame);
-        LERROR(keyTropicalOrbitPeriod << ": " << _tropic);
-        LERROR(keyEarthOrbitRatio << ": " << _ratio);
-        LERROR(keyDayLength << ": " << _day);
-        return false;
+    if (dictionary.hasKeyAndValue<double>(KeyPointSize)) {
+        _pointSize = dictionary.value<double>(KeyPointSize);
     }
+    addProperty(_pointSize);
+
+    _renderingModes.addOptions({
+        { RenderingModeLines, "Lines" },
+        { RenderingModePoints, "Points" },
+        { RenderingModeLinesPoints, "Lines+Points" }
+    });
+
+    if (dictionary.hasKeyAndValue<std::string>(KeyRendering)) {
+        _renderingModes = RenderingModeConversion[
+            dictionary.value<std::string>(KeyRendering)
```

Comment:
Yes; the documentation will check for this with the StringInListVerifier

Added comments to make that clearer